core/amber/src/main/scala/edu/uci/ics/texera/web/service/ExecutionResultService.scala

```diff
@@ -252,22 +252,33 @@ class ExecutionResultService(
                 val sinkMgr = sinkOperators(opId).getStorage
                 if (oldState.resultInfo.isEmpty) {
                   val fields = sinkMgr.getAllFields()
-                  tableFields.update(
-                    opId.id,
-                    Map(
-                      "numericFields" -> fields(0),
-                      "catFields" -> fields(1),
-                      "dateFields" -> fields(2)
+                  if (fields.length >= 3) {
+                    tableFields.update(
+                      opId.id,
+                      Map(
+                        "numericFields" -> fields(0),
```

Comment:
Please make these saved to three `val`, and make the meaning of these constant field names clear.